File: internal/model/tag.go
package model

import "time"

// Tag is a cross-cutting label for categorizing todos.
type Tag struct {
	ID        string    `json:"id" db:"id"`
	Name      string    `json:"name" db:"name"`
	Color     string    `json:"color" db:"color"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

File: Sources/Models/Tag.swift
import Foundation
import SwiftData

@Model
final class Tag {
    var id: UUID
    var name: String
    var color: String

    var todos: [Todo]

    init(
        name: String,
        color: String = "#8E8E93"
    ) {
        self.id = UUID()
        self.name = name
        self.color = color
        self.todos = []
    }
}
